test/functional/test-sanitizer.js

```diff
@@ -174,8 +174,16 @@ describe('sanitizeHtml', () => {
   it('should NOT output blacklisted values for class attributes', () => {
     expect(sanitizeHtml('<p class="i-amphtml-">hello</p>')).to.be
         .equal('<p>hello</p>');
+    expect(sanitizeHtml('<p class="i-amphtml-class">hello</p>')).to.be
+        .equal('<p>hello</p>');
+    expect(sanitizeHtml('<p class="foo-i-amphtml-bar">hello</p>')).to.be
```

Comment:
\W matches the dash between 'foo' and 'i' and then i-amphtml- follows immediately after.